modules/hivemq-edge-module-postgresql/src/main/java/com/hivemq/edge/adapters/postgresql/PostgreSQLPollingProtocolAdapter.java

```diff
@@ -0,0 +1,197 @@
+        return adapterId;
+    }
+
+    @Override
+    public void start(final @NotNull ProtocolAdapterStartInput input, final @NotNull ProtocolAdapterStartOutput output) {
+        try {
+            Class.forName("org.postgresql.Driver");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+
+        /* Test connection to the database when starting the adapter. */
+        try {
+            log.debug("Starting connection to the database instance");
+            databaseConnection = postgreSQLHelpers.connectDatabase(compiledUri, username, password);
+            if(databaseConnection.isValid(0)){
+                output.startedSuccessfully();
+                protocolAdapterState.setConnectionStatus(ProtocolAdapterState.ConnectionStatus.CONNECTED);
+            } else {
+                output.failStart(new Throwable("Error connecting database, please check the configuration"), "Error connecting database, please check the configuration");
+                protocolAdapterState.setConnectionStatus(ProtocolAdapterState.ConnectionStatus.DISCONNECTED);
+            }
+        } catch (final Exception e) {
+            output.failStart(e, null);
+            protocolAdapterState.setConnectionStatus(ProtocolAdapterState.ConnectionStatus.DISCONNECTED);
+        }
+    }
+
+    @Override
+    public void stop(final @NotNull ProtocolAdapterStopInput protocolAdapterStopInput, final @NotNull ProtocolAdapterStopOutput protocolAdapterStopOutput) {
+        try {
+            log.debug("Closing database connection");
+            databaseConnection.close();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        protocolAdapterStopOutput.stoppedSuccessfully();
+    }
+
+
+    @Override
+    public @NotNull ProtocolAdapterInformation getProtocolAdapterInformation() {
+        return adapterInformation;
+    }
+
+    @Override
+    public void poll(final @NotNull PollingInput pollingInput, final @NotNull PollingOutput pollingOutput) {
+        log.debug("Getting polling context");
+        final PollingContext pollingContext = pollingInput.getPollingContext();
+
+        /* Connect to the database and execute the query */
+        try {
+            log.debug("Checking database connection state");
+            if(!databaseConnection.isValid(0)){
+                log.debug("Connecting to the database");
+                databaseConnection = postgreSQLHelpers.connectDatabase(compiledUri, username, password);
+            }
+
+            log.debug("Handling tags for the adapter");
+            tags.stream()
+                    .filter(tag -> tag.getName().equals(pollingContext.getTagName()))
+                    .findFirst()
+                    .ifPresentOrElse(
+                            def -> {
+                                try {
+                                    ResultSet result;
+                                    ObjectMapper om = new ObjectMapper();
```

Comment:
Creation of the ObjectMapper is considerable expensive. As it is thread safe I would suggest to create a static field for it.